src/component/mapping/new-mapping-destination.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import remote from 'remote';
+import createChooseFile from '../../service/choose-file.js';
+
+const {bool, string, func} = React.PropTypes;
+
+const NewMappingTarget = (props) => {
```

Comment:
Hmm, yes and no. I do agree about the `local` bit, not completely happy about that. Otherwise, I wouldn't put this logic up into `NewMapping`, I'd keep this component, but it'd still be an if/else on whether to use `FileMappingDest` or `UrlMappingDest`. Make sure you look at my response to your comment about `NewMapping`, just to clarify things.

And ha, I keep mixing up target and destination in my head, yeah that's a typo.